smart/model.py

```diff
@@ -1158,13 +1164,15 @@ def initialize_discrete_variational_problem_and_solver(self):
         # if use snes
         if self.config.solver["use_snes"]:
             logger.debug("Using SNES solver", extra=dict(format_type="log"))
+            # Does passing the entire model object to smartSNESProblem slow down execution?
```

Comment:
I think one potential problem here is that you are passing in `self` as an object and setting it as an attribute on`self.problem`, meaning that `self.problem.model = self` (=`self.problem.model.problem.model` and so on). This is called a `cycle`, and such an object is never cleaned up which might lead to memory leaks.